include/boost/thread/detail/thread.hpp

```diff
@@ -641,6 +641,12 @@ namespace boost
             thread_data(other.thread_data)
         {}
 
+        id& operator=(const id& other) BOOST_NOEXCEPT
```

Comment:
=default, and same for the copy constructor

Reply:
As noted before, I disagree, because this will affect `noexcept` for some compilers.

Reply:
How relevant is that in practice? A non-noexcept move constructor causes the copy constructor to be used instead, which may be a pessimization (except of course in the case where the move constructor is more expensive, such as when it clears the rhs unnecessarily); but a non-noexcept copy constructor doesn't do anything.

Reply:
It allows `std::swap` to be `noexcept`, for example. It also affects `noexcept` of any default-generated functions in the user's classes that have `id` as a member. Not to mention that `noexcept` is part of the contract. In general, I consider `noexcept` property more important than `=default`.

Reply:
No compiler on Godbolt mis-noexcepts the =default copy. https://godbolt.org/z/esZKLa

Reply:
Hmm, my experience is quite different: https://github.com/boostorg/atomic/blob/dd4cb7e5400b6f69f4fa5dd909146ad170a2a7d5/include/boost/atomic/detail/config.hpp#L94. I'm not sure why godbolt works. Perhaps, older compilers are unstable and the deduction works in some cases and not in other.

Reply:
I think that your configuration macro refers to the case
```
id( id const& ) noexcept = default;
```
which is indeed not supported by some early versions.

The case
```
id( id const& ) = default;
```
is universally supported, and the only case in which noexcept is not deduced correctly (on Godbolt) is on Intel 13, when there is a nontrivial member with a noexcept copy. https://godbolt.org/z/LhTy_A

We can ifdef this out in principle (this only affects the smart pointer case), but how important is Intel 13 nowadays? 16 and higher all work.

Reply:
I decided to just remove the copy constructor to allow the compiler generate the full set of constructors/assignment operators. Hopefully, it will do the right thing.